lib/facter/ds389__instances.rb

```diff
@@ -43,19 +54,62 @@
       slapd_svr_conf = File.join(slapd_svr, 'dse.ldif')
 
       if File.exist?(slapd_svr_conf)
-        instance_name = File.basename(slapd_svr)
-        slapd_svr_conf_content = File.read(slapd_svr_conf).lines
-
-        slapd_svr_conf_content.grep(%r{nsslapd-(port|listenhost):\s}).each do |config_item|
-          key, value = config_item.split(': ')
-
-          if key.include?('-listenhost')
-            instances[instance_name] ||= {}
-            instances[instance_name]['address'] = value.to_i
-          elsif key.include?('-port')
-            instances[instance_name] ||= {}
-            instances[instance_name]['port'] = value.to_i
+        instance_name = File.basename(slapd_svr).split('slapd-').last
```

Comment:
They're LDIFs, not INI files. Techincally, there's an augeas lens for them in place but I didn't care to figure out how it works since I only needed (relatively) easily parseable stuff. If it gets more complex, I'll probably look at the lens instead.